.github/workflows/validate.yml

```diff
@@ -0,0 +1,37 @@
+name: validate
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - master
```

Comment:
Perhaps we should already add `main` as well in preparation of renaming at some point.